src/map/status.c

```diff
@@ -8723,40 +8723,186 @@ static int status_change_start_sub(struct block_list *src, struct block_list *bl
+						flag |= SCFLAG_NOICON;
+				}
+				if (sum_water <= 0) {
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_WATER));
+					flag |= SCFLAG_NOICON;
+				}
+
+				// Wind
+				int sum_wind = val2 + ((sce_wind != NULL) ? sce_wind->val1 : 0);
+				sum_wind += (sce_all != NULL) ? sce_all->val4 : 0;
+				if (sum_wind <= 0)
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_WIND));
+
+				break;
+			}
+			case SC_RESIST_PROPERTY_GROUND: {
+				struct status_change_entry *sce_all = sc->data[SC_ARMOR_RESIST];
+				struct status_change_entry *sce_wind = sc->data[SC_RESIST_PROPERTY_WIND];
+				struct status_change_entry *sce_fire = sc->data[SC_RESIST_PROPERTY_FIRE];
+
+				// Ground
+				int sum_ground = val1 + ((sce_wind != NULL) ? sce_wind->val2 : 0);
+				if (sce_all != NULL) {
+					sum_ground += sce_all->val2;
+					int left = (int)DIFF_TICK(timer->get(sce_all->timer)->tick, timer->gettick());
+					if (left > total_tick && sum_ground - val1 > 0)
+						flag |= SCFLAG_NOICON;
+				}
+				if (sum_ground <= 0) {
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_GROUND));
+					flag |= SCFLAG_NOICON;
+				}
+
+				// Fire
+				int sum_fire = val2 + ((sce_fire != NULL) ? sce_fire->val1 : 0);
+				sum_fire += (sce_all != NULL) ? sce_all->val3 : 0;
+				if (sum_fire <= 0)
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_FIRE));
+
+				break;
+			}
+			case SC_RESIST_PROPERTY_FIRE: {
+				struct status_change_entry *sce_all = sc->data[SC_ARMOR_RESIST];
+				struct status_change_entry *sce_ground = sc->data[SC_RESIST_PROPERTY_GROUND];
+				struct status_change_entry *sce_water = sc->data[SC_RESIST_PROPERTY_WATER];
+
+				// Fire
+				int sum_fire = val1 + ((sce_ground != NULL) ? sce_ground->val2 : 0);
+				if (sce_all != NULL) {
+					sum_fire += sce_all->val3;
+					int left = (int)DIFF_TICK(timer->get(sce_all->timer)->tick, timer->gettick());
+					if (left > total_tick && sum_fire - val1 > 0)
+						flag |= SCFLAG_NOICON;
+				}
+				if (sum_fire <= 0) {
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_FIRE));
 					flag |= SCFLAG_NOICON;
+				}
+
+				// Water
+				int sum_water = val2 + ((sce_water != NULL) ? sce_water->val1 : 0);
+				sum_water += (sce_all != NULL) ? sce_all->val1 : 0;
+				if (sum_water <= 0)
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_WATER));
+
+				break;
+			}
+			case SC_RESIST_PROPERTY_WIND: {
+				struct status_change_entry *sce_all = sc->data[SC_ARMOR_RESIST];
+				struct status_change_entry *sce_water = sc->data[SC_RESIST_PROPERTY_WATER];
+				struct status_change_entry *sce_ground = sc->data[SC_RESIST_PROPERTY_GROUND];
+
+				// Wind
+				int sum_wind = val1 + ((sce_water != NULL) ? sce_water->val2 : 0);
+				if (sce_all != NULL) {
+					sum_wind += sce_all->val4;
+					int left = (int)DIFF_TICK(timer->get(sce_all->timer)->tick, timer->gettick());
```

Comment:
``timer->get`` may return NULL. also sce_all->timer can be invalid timer. probably need add checks for timer field and return value of ``timer->get``?